### ISSUE TITLE
question about reward score

### ISSUE THREAD

**Issue by DecideToLeave** [Edward-Sun/easy-to-hard#7]
Why is the score the average of the dimensions in the vocabulary table instead of taking the corresponding logits at the corresponding position ?

thank you

**Comment by Edward-Sun** [comment]
Hi @DecideToLeave 

This is a design when we export the model to the huggingface format, so that we don't have to change the original model config.

During training, we only have two logits in the last layer. So essentially, the logits would look like ababababab...

https://github.com/Edward-Sun/easy-to-hard/blob/24af9789e80b09a3b33af1fe5e02b5048578c3bf/scripts/convert_checkpoint_to_hf.py#L91

**Comment by DecideToLeave** [comment]
So in the "\n\n" position
`Rewards=rewards. mean (dim=-1)`
Is it actually calculated based only on the logits of a and b ？

**Comment by Edward-Sun** [comment]
Correct.

**Comment by DecideToLeave** [comment]
thanks, and what is the difference between using only the logits of a or b for binary classification and calculating the cross entropy based on the mean of both.

**Comment by Edward-Sun** [comment]
If you want to use binary classification, you should not use a or b, but a or -b (or maybe a/2 or -b/2).

There is no difference between applying a logisitic function to the mean of both or using a binary classification. The results are numeircally the same.

**Comment by DecideToLeave** [comment]
I'm a bit confused about why it's a or -b. What I understand is that if there are only two logits in the last layer, is it similar to the first logits representing negative sample scores and the second logits representing positive sample scores? So, I can calculate the loss based on the probability after the corresponding softmax using the 0/1 index.
